tiledb/sm/serialization/array.cc

```diff
@@ -358,6 +365,246 @@ Status array_open_from_capnp(
+      array_fragments_list_builder->initEntries(fragments.size());
+  for (size_t i = 0; i < fragments.size(); i++) {
+    const auto& relative_uri = serialize_array_uri_to_relative(fragments[i]);
+    entries_builder.set(i, relative_uri);
+  }
+}
+
+std::vector<URI> fragments_list_from_capnp(
+    const capnp::ArrayFragmentsList::Reader& array_fragments_list_reader,
+    const URI& array_uri) {
+  if (array_fragments_list_reader.hasEntries()) {
+    std::vector<URI> fragments;
+    fragments.reserve(array_fragments_list_reader.getEntries().size());
+    for (auto entry : array_fragments_list_reader.getEntries()) {
+      fragments.emplace_back(
+          deserialize_array_uri_to_absolute(entry.cStr(), array_uri));
+    }
+    return fragments;
+  } else {
+    throw ArraySerializationException(
+        "[fragments_list_from_capnp] There are no fragments to deserialize");
+  }
+}
+
+void fragments_list_serialize(
+    const std::vector<URI>& fragments,
+    SerializationType serialize_type,
+    Buffer* serialized_buffer) {
+  if (fragments.empty()) {
+    throw ArraySerializationException(
+        "[fragments_list_serialize] Fragments vector is empty");
+  }
+
+  try {
+    // Serialize
+    ::capnp::MallocMessageBuilder message;
+    auto builder = message.initRoot<capnp::ArrayFragmentsList>();
+    fragments_list_to_capnp(fragments, &builder);
+
+    // Copy to buffer
+    serialized_buffer->reset_size();
+    serialized_buffer->reset_offset();
+    switch (serialize_type) {
+      case SerializationType::JSON: {
+        ::capnp::JsonCodec json;
+        kj::String capnp_json = json.encode(builder);
+        const auto json_len = capnp_json.size();
+        const char nul = '\0';
+        // size does not include needed null terminator, so add +1
+        throw_if_not_ok(serialized_buffer->realloc(json_len + 1));
+        throw_if_not_ok(serialized_buffer->write(capnp_json.cStr(), json_len));
+        throw_if_not_ok(serialized_buffer->write(&nul, 1));
+        break;
+      }
+      case SerializationType::CAPNP: {
+        kj::Array<::capnp::word> protomessage = messageToFlatArray(message);
+        kj::ArrayPtr<const char> message_chars = protomessage.asChars();
+        const auto nbytes = message_chars.size();
+        throw_if_not_ok(serialized_buffer->realloc(nbytes));
+        throw_if_not_ok(
+            serialized_buffer->write(message_chars.begin(), nbytes));
+        break;
+      }
+      default: {
+        throw ArraySerializationException(
+            "[fragments_list_serialize] Unknown serialization type passed");
+      }
+    }
+
+  } catch (kj::Exception& e) {
+    throw ArraySerializationException(
+        "[fragments_list_serialize] kj::Exception: " +
+        std::string(e.getDescription().cStr()));
+  } catch (std::exception& e) {
+    throw ArraySerializationException(
+        "[fragments_list_serialize] exception " + std::string(e.what()));
+  }
+}
+
+std::vector<URI> fragments_list_deserialize(
```

Comment:
Please don't hate me, would you mind putting this fragments URIs serialization/deserialization code in its own file, e.g. serialization/fragments.cc?